File: Gas Station/answer_slow.cpp
// O(n^2) solution
// Brute and Force

class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        if(gas.empty())
            return -1;
        for(int i=0;i<gas.size();i++)
        {
            int sum=0;
            for(int j=0;j<gas.size();j++)
            {
                int x = (i+j)%gas.size();
                sum = sum + gas[x] - cost[x];
                if(sum<0)
                    break;
            }
            if(sum>=0)
                return i;
        }
        return -1;
    }
};

File: Subarray Sum Equals K/answer_slow.cs
// a brute and force O(n^2) solution
// if it is a C++ implementation, it would meet a TLE error.

public class Solution {
    public int SubarraySum(int[] nums, int k) {
        int ret = 0;
        if(nums.Length == 0)
            return 0;
            
        for(int i=0;i<nums.Length;i++)
        {
            int sum = 0;
            for(int j=i;j<nums.Length;j++)
            {
                sum += nums[j];
                if(sum == k)
                    ret++;
            }
        }
        return ret;
    }
}
